src/compiler/interface.jl

```diff
@@ -65,11 +63,14 @@ Base.adjoint(f::Function) = x -> gradient(f, x)[1]
 
 # TODO store ids only
 struct Params
-  order::Buffer{Any, Vector{Any}}
+  order::Buffer # {Any, Vector{Any}}
   params::IdSet{Any}
-  Params() = new(Buffer([], false), IdSet())
 end
 
+Params() = Params(Buffer([], false), IdSet())
+Params(xs::Vector) = Params(Buffer(xs, false), IdSet(xs))
```

Comment:
```suggestion
Params(xs::Vector) = Params(Buffer(xs, false), IdSet(xs))
Params(xs::Tuple) = Params(Buffer(xs, false), IdSet(xs))
```

Alternatively, go back to the old constructor with `push!` and `@nograd` for `Params`.

Reply:
Ah, those are in fact using tuples, but that's alright I think, those are invariant anyway. The issue is an implicit call to collect. I think to maintain compatibility it would mean to support that call to collect on tuples.

We can nograd the constructors too